### ISSUE TITLE
SBOM generation for release PRs

### ISSUE THREAD

**Issue by lance** [google-github-actions/release-please-action#794]
### TL;DR

I am trying to generate an SBOM for my npm module upon release. However, there is a bit of a chicken and egg problem. When the release PR is merged, this includes changes to the package-lock.json file which in turn results in a modification of the SBOM, which means that when the release is published, the SBOM is out of date.

A tangible example:

- clone this repository: https://github.com/nodeshift/faas-js-runtime
- change into the repo directory and run `git checkout v2.2.1` (the most recent release tag)
- run `npm run sbom` which generates an SBOM based on the current `package-lock.json` file
- run `git diff` to see what should have been included in the release SBOM

<details>
<summary>SBOM diff</summary>

```diff
diff --git a/sbom.json b/sbom.json
index 8c54a78..3a71768 100644
--- a/sbom.json
+++ b/sbom.json
@@ -3,9 +3,9 @@
   "bomFormat": "CycloneDX",
   "specVersion": "1.4",
   "version": 1,
-  "serialNumber": "urn:uuid:1a3adee4-7f98-4b19-8c50-97eda0e1d650",
+  "serialNumber": "urn:uuid:c6ddbb8b-facb-47bb-ac8e-ea46ecb8a489",
   "metadata": {
-    "timestamp": "2023-07-13T21:02:27.796Z",
+    "timestamp": "2023-07-13T21:56:55.035Z",
     "tools": [
       {
         "vendor": "@cyclonedx",
@@ -55,8 +55,8 @@
     "component": {
       "type": "application",
       "name": "faas-js-runtime",
-      "version": "2.2.0",
-      "bom-ref": "faas-js-runtime@2.2.0",
+      "version": "2.2.1",
+      "bom-ref": "faas-js-runtime@2.2.1",
       "author": "Red Hat, Inc.",
       "description": "A Node.js framework for executing arbitrary functions in response to HTTP or cloud events",
       "licenses": [
@@ -66,7 +66,7 @@
           }
         }
       ],
-      "purl": "pkg:npm/faas-js-runtime@2.2.0?vcs_url=git%2Bhttps%3A//github.com/nodeshift/faas-js-runtime.git",
+      "purl": "pkg:npm/faas-js-runtime@2.2.1?vcs_url=git%2Bhttps%3A//github.com/nodeshift/faas-js-runtime.git",
       "externalReferences": [
         {
           "url": "git+https://github.com/nodeshift/faas-js-runtime.git",
@@ -2764,7 +2764,7 @@
   ],
   "dependencies": [
     {
-      "ref": "faas-js-runtime@2.2.0",
+      "ref": "faas-js-runtime@2.2.1",
       "dependsOn": [
         "cloudevents@7.0.1",
         "commander@11.0.0",
```
</details>

It seems that it may be possible to do this with plugins (https://github.com/googleapis/release-please/blob/main/docs/manifest-releaser.md#plugins). But it's not clear or straightforward.

### Detailed design

As a part of the PR generation, `release-please` could use the `@cyclonedx/cyclonedx-library` to optionally generate an SBOM and commit the changes.

### Additional information

A stopgap solution may be to create a draft release and manually update it with modifications to the SBOM but this will need to happen with each update to the release PR.

**Comment by stevespringett** [comment]
With CycloneDX, it would also be possible to include machine-readable release notes as well. See https://cyclonedx.org/docs/1.5/json/#metadata_component_releaseNotes for info.

**Comment by lance** [comment]
Closing this as out of scope based on https://github.com/googleapis/release-please/issues/2014#issuecomment-1650585387

**Comment by lance** [comment]
Note that I have a workaround with https://github.com/nodeshift/faas-js-runtime/blob/6a3b33a660f84af721c0adae27acab861f1024e5/.github/workflows/release-please.yaml#L21-L29. It may be nice to add something like this to the README or other documentation, as I am sure that this will be a common workflow.
